src/components/atomics/ASwitch.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <SwitchGroup as="div" class="alias-flex alias-items-center alias-pt-2 alias-pb-2">
+    <SwitchLabel
+        v-if="label"
+        as="span"
+        class="alias-mr-3"
+    >
+      <span class="alias-block alias-text-xs alias-font-medium alias-text-gray-900">{{ label }}</span>
+    </SwitchLabel>
+    <Switch
+        v-model="checked"
+        @update:checked="update"
+        :class="[checked ? 'alias-bg-primary-500' : 'alias-bg-gray-200']"
+        class="alias-relative alias-inline-flex alias-w-[74px] alias-shrink-0 alias-cursor-pointer alias-rounded-full
+        alias-border-2 alias-border-transparent alias-transition-colors alias-duration-200 alias-ease-in-out
+        focus:alias-outline-none focus:alias-ring-2 focus:alias-ring-offset-2 focus:alias-ring-primary-500"
+    >
+      <span aria-hidden="true"
+          :class="[checked ? 'alias-translate-x-9' : 'alias-translate-x-0']"
+          class="alias-pointer-events-none alias-inline-block alias-h-[34px] alias-w-[34px]
+          alias-transform alias-rounded-full alias-bg-white alias-shadow alias-ring-0 alias-transition
+          alias-duration-200 alias-ease-in-out"
+    />
+    </Switch>
+  </SwitchGroup>
+</template>
+
+<script setup lang="ts">
+import {Switch, SwitchGroup, SwitchLabel} from '@headlessui/vue'
+import { PropType, toRefs } from 'vue'
+import { ColorNames } from '../../AliasColors'
+
+const props = defineProps({
+  type: { type: String as PropType<ColorNames>, required: true },
+  label: {
+    type: String,
+    default: ''
+  },
+  checked: {
+    type: Boolean,
+    required: true
+  }
+})
```

Comment:
could you add here some JSDoc so we know, in storybook and in the project that will install the alias framework, what are this props for ? 
If you want an example, you can check this file https://github.com/progressive-identity/alias-ui-framework/blob/staging/src/components/atomics/AButton.vue

JSDoc is presented like that

```
/**
   * The color of the button
   * The possible values are listed in {@link AliasColorNames}
   */
```

Reply:
thanks Jules for ur feedbacks, i added the comments for the props and i added the component to the index.ts file. sorry for the oversight.